internal/metamorph/processor.go

```diff
@@ -86,6 +86,14 @@ type Processor struct {
 
 	processMinedInterval  time.Duration
 	processMinedBatchSize int
+
+	announcedTransactions     []AnnouncedTransaction
```

Comment:
Please add a comment that the collection is ordered, as it will make it much easier to understand how the code below works.
Or change name to smt like "orderedAnnouncedTxsCollection"